test/one/profiler/test/Runner.java

```diff
@@ -206,21 +233,35 @@ public static void main(String[] args) throws Exception {
         configureLogging();
         configureSkipTests();
 
-        boolean passedAll = true;
-        for (String arg : args) {
-            String testName = arg;
-            if (testName.indexOf('.') < 0 && Character.isLowerCase(testName.charAt(0))) {
-                // Convert package name to class name
-                testName = "test." + testName + "." + Character.toUpperCase(testName.charAt(0)) + testName.substring(1) + "Tests";
+        List<RunnableTest> allTests = getRunnableTests(args);
+        final int testCount = allTests.size();
+
+        int i = 1;
+        long totalTestDuration = 0;
+        Map<TestStatus, Integer> statusCounts = new HashMap<>();
+        List<String> failedTests = new ArrayList<>();
+        for (RunnableTest rt : allTests) {
+            long start = System.nanoTime();
+            TestResult result = run(rt);
+            long durationNs = System.nanoTime() - start;
+
+            statusCounts.put(result.status(), statusCounts.getOrDefault(result.status(), 0) + 1);
+            totalTestDuration += durationNs;
+            String message = formatTestMessage(rt, result, i, testCount, durationNs);
+            if (result.status() == TestStatus.FAIL) {
+                failedTests.add(rt.testInfo());
             }
-            passedAll &= run(Class.forName(testName));
+            System.out.println(message);
```

Comment:
Runner only used logging before your change.
I don't mind println, but let's then revise existing messages: which should remain optional logs and which should be always printed.

Reply:
I’ve kept only `"Running " + rt.testInfo() + "..."` as an info log, while all other outputs are directed to stdout via println.

Are there any specific println statements you think should be made optional?

To further reduce the output, we could revisit this in a separate PR. A possible enhancement would be to introduce a verbosity flag. Runner would captures all stdout/stderr output from tests themselves and print them only when the flag is enabled. By default, the runner would print only "Running...", the test result (PASS/FAIL/SKIP), and a summary at the end of all tests.

Reply:
OK, agreed. Let's leave it as is.